test/integration/image_test.go

```diff
@@ -0,0 +1,132 @@
+	ctx, cancel := context.WithTimeout(context.Background(), Minutes(15))
+	defer Cleanup(t, profile, cancel)
+
+	// Serial tests
+	t.Run("serial", func(t *testing.T) {
+		tests := []struct {
+			name      string
+			validator validateFunc
+		}{
+			{"NormalBuild", validateNormalImageBuild},
+			{"BuildWithBuildArg", validateImageBuildWithBuildArg},
+			{"BuildWithDockerIgnore", validateImageBuildWithDockerIgnore},
+			{"BuildWithSpecifiedDockefile", validateNormalImageBuildWithSpecifiedDockefile},
+		}
+		for _, tc := range tests {
+			if ctx.Err() == context.DeadlineExceeded {
+				t.Fatalf("Unable to run more tests (deadline exceeded)")
+			}
+			if t.Failed() {
+				t.Fatalf("Previous test failed, not running dependent tests")
+			}
+
+			t.Run(tc.name, func(t *testing.T) {
+				tc.validator(ctx, t, profile)
+				if t.Failed() && *postMortemLogs {
+					PostMortemLogs(t, profile)
+				}
+			})
+		}
+	})
+}
+
+// normal test case for minikube image build, with -t parameter
+func validateNormalImageBuild(ctx context.Context, t *testing.T, profile string) {
+	defer PostMortemLogs(t, profile)
+	args := append([]string{"image", "build", "-t", "aaa:latest", "./testdata/image-build/test-normal", "-p", profile}, StartArgs()...)
+	rr, err := Run(t, exec.CommandContext(ctx, Target(), args...))
+	if err != nil {
+		t.Fatalf("failed to start minikube with args: %q : %v", rr.Command(), err)
+	}
+}
+
+// normal test case for minikube image build, with -t and -f parameter
+func validateNormalImageBuildWithSpecifiedDockefile(ctx context.Context, t *testing.T, profile string) {
+	defer PostMortemLogs(t, profile)
+	args := append([]string{"image", "build", "-t", "aaa:latest", "-f", "inner/Dockerfile", "./testdata/image-build/test-f", "-p", profile}, StartArgs()...)
+	rr, err := Run(t, exec.CommandContext(ctx, Target(), args...))
+	data, _ := io.ReadAll(rr.Stdout)
+
+	output := string(data)
+	fmt.Println(output)
+	if err != nil {
+		t.Fatalf("failed to start minikube with args: %q : %v", rr.Command(), err)
+	}
+}
+
+// test case for bug in https://github.com/kubernetes/minikube/issues/12384
+// build with --build-opt
+func validateImageBuildWithBuildArg(ctx context.Context, t *testing.T, profile string) {
+	defer PostMortemLogs(t, profile)
+	args := append([]string{"image", "build", "-t", "aaa:latest", "--build-opt=build-arg=ENV_A=aaasdftggdf", "--build-opt=no-cache", "./testdata/image-build/test-arg", "-p", profile}, StartArgs()...)
+	rr, err := Run(t, exec.CommandContext(ctx, Target(), args...))
+	data, _ := io.ReadAll(rr.Stdout)
+
+	output := string(data)
+	if err != nil {
+		t.Fatalf("failed to build image with args: %q : %v", rr.Command(), err)
+	}
+	if !strings.Contains(output, "aaasdftggdf") {
+		t.Fatalf("failed to pass build-args with args: %q : %v", rr.Command(), err)
+	}
+}
+
+// current this test cannot be passed because issue https://github.com/kubernetes/minikube/issues/12431 hasn't been fixed, so this test case is not enabled
```

Comment:
It makes sense to skip the test, but the way it's being skipped makes it hard to notice it's being skipped without reading the comment. I'd recommend you add the test to test list above and add the following to the first line of the func below.
```
t.Skip("skipping due to https://github.com/kubernetes/minikube/issues/12431")
```

It will show up as being skipped in test output.